### PR TITLE
[WIP] proper esdocs implementation

### PR DESCRIPTION
**Pull request by axe312ger**
An updated version of the reference docs, for comparision with the current one, can be found here:
https://axe312ger.github.io/contentful-management.js/

### CODE REVIEW

File: lib/entities/asset.js [hunk 1]
```diff
@@ -117,14 +117,14 @@ const ASSET_PROCESSING_CHECK_RETRIES = 5
...
  * // require contentful-management
  * var contentfulManagement = require('contentful-management')
  * var client = contentfulManagement.createClient({
  * // This is the access token for this space. Normally you get both ID and the token in the Contentful web app
  * accessToken: 'YOUR_ACCESS_TOKEN'
- * }) 
- * 
+ * })
+ *
  * let space = await client.getSpace('SPACE_ID')
```
Comment (TimBeyer) [inline lib/entities/asset.js]: Not really part of this PR but I feel using `await` is a bit premature since most people won't be able to run it just like that. I think we shouldn't use it in examples until it's widely available.

Comment (Khaledgarbaya) [inline lib/entities/asset.js]: Yes @TimBeyer you are right, this was me introducing this in the CDA documentations. We'll work on making all the examples using promises.

File: lib/create-contentful-api.js [hunk 2]
```diff
@@ -1,73 +1,73 @@
 import errorHandler from './error-handler'
 import entities from './entities'
 
+const {wrapSpace, wrapSpaceCollection} = entities.space
 
 /**
- * @typedef {ClientAPI} ClientAPI
```
Comment (Khaledgarbaya) [inline lib/create-contentful-api.js]: I think we can merge this with `ContentfulManagement` Object as it sound like there is a duplicate there

File: lib/create-contentful-api.js [hunk 3]
```diff
@@ -1,73 +1,73 @@
...
+/**
+ * Creates API object with methods to access functionality from Contentful's
+ * Management API
+ *
+ * @param {Object} params - API initialization params
+ * @param {Object} params.http - HTTP client instance
+ * @param {Function} params.shouldLinksResolve - Link resolver preconfigured with global setting
+ * @return {ClientAPI}
+ */
+export default function createClientAPI (params) {
```
Comment (Khaledgarbaya) [inline lib/create-contentful-api.js]: `createClientAPI` should be `@private` so we should not expose in the documentation, as it will be confusing with `createClient`

**Comment by axe312ger** [comment]
Closing that one since we won't switch to esdocs
